### DISCUSSION TITLE
KolHeading: _label oder children?

### DISCUSSION THREAD

**Discussion by Chrisdo82** [public-ui/kolibri#2493] [Q&A] [answered]
Mit der letzten Version 1.4.0 wurde das `_label` bei der `KolHeading` verpflichtend. Trotzdem können `children` übergeben werden. Vorher sah der Code bei uns so aus:

```
<KolHeading _level={3} className="flex items-center">
    {badgeNumber !== undefined && (
        <span className="mr-4 inline-flex h-7 w-7 items-center justify-center rounded-full bg-midnight text-base font-bold text-white">
            {badgeNumber}
        </span>
    )}
    <span className="text-heading">{headline}</span>
</KolHeading>
```

Da dem `_label` nur Strings übergeben werden können, sehe ich derzeit nur 2 Möglichkeiten:

1. ich übergebe _label einen leeren String (`_label=""`) und lasse die die `children` wie sie sind
2. ich übergebe gar keine `children` mehr:
```
<div className="flex items-center">
    {badgeNumber !== undefined && (
        <span className="mr-4 inline-flex h-7 w-7 items-center justify-center rounded-full bg-midnight text-base font-bold text-white">
            {badgeNumber}
        </span>
    )}
    <KolHeading _level={3} _label={headline} className="text-heading" />
</div>
```

Wenn 2. der präferierte Weg ist, warum können dann nach wie vor `children` übergeben werden?

**Comment by deleonio** [comment, marked as answer]
Hi @Chrisdo82 ,

die Heading-, wie auch die Link-Komponente wurde für den Expert-Slot vorbereitet. Da dieser aber bisher nicht verpflichtend war, geht der Slot unverändert. Lediglich die Angabe eines leeren `_label` wird verpflichtend.

Aktuelle Liste: https://public-ui.github.io/docs/konzepte/expert-slot#komponentenliste
